.github/workflows/ci.yml

```diff
@@ -225,22 +260,72 @@ jobs:
         run: |
           git config --unset-all '${{ steps.setup_git_auth.outputs.config_key }}'
           git submodule foreach git config --unset-all '${{ steps.setup_git_auth.outputs.config_key }}'
+                    
+          echo ${{ secrets.PAT_REVIEW }} | gh auth login --with-token
+
+      - id: create_pr
+        name: Create Pull Request if score changed, if not cleanup branch and PR
+        if: startsWith(github.event_name, 'pull_request')
+        run: |
+          base_repo_owner=${{ github.event.pull_request.base.repo.owner.login }}
+          base_repo_name=${{ github.event.pull_request.base.repo.name }}
+          challenge_scores_branch=${{steps.push_challenge_scores.outputs.challenge_scores_branch}}
+
+          pr_number=$(gh api repos/$base_repo_owner/$base_repo_name/pulls -q ".[] | select(.head.ref == \"$challenge_scores_branch\") | .number")
+
+          if [ -z "${{steps.push_challenge_scores.outputs.updated_score}}" ]; then
+            if [ -n "$pr_number" ]; then
+              gh api -X PATCH repos/$base_repo_owner/$base_repo_name/pulls/$pr_number -f state=closed
+              echo "Close PR $pr_number if not already closed"
+            fi
+            # Delete branch
+            gh api -X DELETE repos/$base_repo_owner/$base_repo_name/git/refs/heads/$challenge_scores_branch \
+              &&  echo "Branch $challenge_scores_branch deleted." \
+              || echo "Branch $challenge_scores_branch not found or already deleted."
+            echo "Score didn't change."
+            exit 0
+          fi
+
+          if [ -z "$pr_number" ] || [ "$(gh api repos/$base_repo_owner/$base_repo_name/pulls/$pr_number --jq ".state")" = "closed" ]; then
+            pr_url=$(gh pr create --title "Update Score In Pull Request Number ${{ github.event.pull_request.number }}" \
+              --head "$base_repo_owner:$challenge_scores_branch" \
+              --base "${{ github.event.pull_request.head.ref }}" \
+              --body "This pull request updates the challenge scores of Auto-GPT. Please check the files changed and merge the pull request." \
+              --repo ${{ github.event.pull_request.head.repo.full_name }})
+          else
+            pr_url=$(gh pr view $pr_number --json url --jq '.url')
+          fi
+          
+          echo "pr_url=$pr_url" >> $GITHUB_OUTPUT  
```

Comment:
Have you tried using an off the shelf action for this? https://github.com/marketplace/actions/create-pull-request